File: src/lib/api/chat_tools.ts
import { invoke } from "@tauri-apps/api/core";

/**
 * 聊天工具权限配置
 */
export interface ChatToolsConfig {
  /** 被禁用的 MCP Server 列表 */
  disabled_servers: string[];
  /** 每个启用的 Server 中被禁用的工具列表 */
  disabled_tools: Record<string, string[]>;
}

/**
 * 获取指定 accountId + sessionId 的工具权限配置
 */
export async function getChatToolsConfig(
  accountId: string,
  sessionId: string
): Promise<ChatToolsConfig> {
  return invoke<ChatToolsConfig>("get_chat_tools_config", {
    accountId,
    sessionId,
  });
}

/**
 * 保存指定 accountId + sessionId 的工具权限配置
 */
export async function saveChatToolsConfig(
  accountId: string,
  sessionId: string,
  config: ChatToolsConfig
): Promise<void> {
  return invoke<void>("save_chat_tools_config", {
    accountId,
    sessionId,
    config,
  });
}

/**
 * 删除指定 accountId + sessionId 的工具权限配置（恢复默认）
 */
export async function deleteChatToolsConfig(
  accountId: string,
  sessionId: string
): Promise<void> {
  return invoke<void>("delete_chat_tools_config", {
    accountId,
    sessionId,
  });
}

File: src-tauri/src/commands/chat_tools.rs
//! 聊天工具权限 Command 层
//!
//! 薄包装层 - 仅做参数转发和错误转换，业务逻辑在 services::chat_tools_service 中。

use std::sync::Arc;

use crate::provider::cache::Cache;
use crate::services::chat_tools_service::{self, ChatToolsConfig};
use tauri::State;

/// 获取指定 accountId + sessionId 的工具权限配置
#[tauri::command]
pub async fn get_chat_tools_config(
    cache: State<'_, Arc<Cache>>,
    account_id: String,
    session_id: String,
) -> Result<ChatToolsConfig, String> {
    Ok(chat_tools_service::get_tools_config(
        Arc::clone(&cache),
        account_id,
        session_id,
    ))
}

/// 保存指定 accountId + sessionId 的工具权限配置
#[tauri::command]
pub async fn save_chat_tools_config(
    cache: State<'_, Arc<Cache>>,
    account_id: String,
    session_id: String,
    config: ChatToolsConfig,
) -> Result<(), String> {
    chat_tools_service::set_tools_config(Arc::clone(&cache), account_id, session_id, config)
}

/// 删除指定 accountId + sessionId 的工具权限配置（恢复默认）
#[tauri::command]
pub async fn delete_chat_tools_config(
    cache: State<'_, Arc<Cache>>,
    account_id: String,
    session_id: String,
) -> Result<(), String> {
    chat_tools_service::remove_tools_config(Arc::clone(&cache), account_id, session_id)
}
